pkg/controllers/cloudprovideraccount/controller.go

```diff
@@ -137,14 +138,15 @@ func (r *CloudProviderAccountReconciler) processCreateOrUpdate(namespacedName *t
 	r.Log.Info("Received request", "account", namespacedName, "operation", "create/update")
 	accountCloudType, err := util.GetAccountProviderType(account)
 	if err != nil {
-		return fmt.Errorf("%s: %v", accountmanager.ErrorMsgAddOrUpdateAccount, err)
+		return fmt.Errorf("failed to add or update account: %v", err)
 	}
 	if err = r.AccManager.AddAccount(namespacedName, accountCloudType, account); err != nil {
 		account.Status.Error = err.Error()
 		r.Log.Info("Setting account status", "account", namespacedName, "err", err)
 		if err = r.Client.Status().Update(context.TODO(), account); err != nil {
 			return fmt.Errorf("failed to update account status, account %v err %v", namespacedName, err)
 		}
+		_ = r.AccManager.RemoveAccount(namespacedName)
```

Comment:
do we really need to call removal? the AddAccount will be retried right?

Reply:
Yes. We don't want to reconciler to keep trying whenever there is an error returned by CPA. 
If for any reason, we are not able to process CPA request, we will set the error status on the CPA object and clean up our internal state.